src/main/java/io/confluent/connect/jdbc/sink/JdbcSinkTask.java

```diff
@@ -98,6 +113,14 @@ public void put(Collection<SinkRecord> records) {
     remainingRetries = config.maxRetries;
   }
 
+  private void retryAndSendDLQ(Collection<SinkRecord> records) {
+    try {
+      writer.write(records);
```

Comment:
According to https://confluentinc.atlassian.net/browse/CC-9435, it looks like suggestions from SEs on the FF tickets indicate this approach. But, to elaborate, this implicit retry before sending to the DLQ is because the sink DLQ is designed to send one record and its corresponding error at a time; since connectors such as the JDBC sink connector often send records as batches with one sort of batch exception in response to any error, it is very difficult to know which record caused errors and if multiple records, which errors correspond to which record. This one time retry allows for all the records in the erred batch to retry to see which had an error, and send those to the DLQ.

Reply:
Thanks for the context @aakashnshah . I think that makes sense, especially since a failed record in the batch may cause records following it to not be delivered without any indication of this to the caller. Let's just make sure that we don't implicitly retry on failure when the user doesn't have DLQ configured though. I see that you are relying on the fact that framework would give us null `ErrantRecordReporter` for this. Let's have some tests to verify this as well.